### DISCUSSION TITLE
为什么执行vela workflow restart app之后，要等半个小时才它真正执行workflow？

### DISCUSSION THREAD

**Discussion by miyoosan** [kubevela/kubevela#4838] [Show and tell] [answered]
问题：执行`vela workflow restart app`之后，半个小时后它才执行，执行起来也很慢。为什么呢？

在雾雾的帮助排查下，找到了原因！

原来，`vela-core`的启动参数`application-re-sync-period`默认值为`5min`, 在这个情况下，如果某个应用`Reconciliation`过程中失败，它会一直重试，导致积压了大量的此类事件，甚至让整个vela-core都没法即时响应，对每次事件的处理都延后了半个小时。

于是，我们就可以看到，`restart workflow`好像根本没反应，或者apply了一个新的app，它的工作流也是没有任何状态的。

解决办法很简单，可以执行`kubectl edit deploy kubevela-vela-core -n vela-system -o yaml`，调大`application-re-sync-period`参数的值，比如`10h`，保存退出, vela-core会自动重启。先等待一段时间，比如半个小时，让`vela-core`消化完所有事件，再找出原因，解决它，就OK了。

**Comment by miyoosan** [comment, marked as answer]
`application-re-sync-period`的意义参见文档[启动参数说明]：https://kubevela.net/zh/docs/platform-engineers/system-operation/bootstrap-parameters

**Reply by FogDong** [reply]
We should add docs for `application-re-sync-period`. cc @wonderflow  @Somefive

**Reply by FogDong** [reply]
Ref: https://kubevela.net/zh/docs/next/platform-engineers/workflow/working-mechanism#%E7%8A%B6%E6%80%81%E7%BB%B4%E6%8C%81
